TimePicker (mobile): Implement 4-15 TimePicker dialog(SIP)

HunseopJeong opened the pull request:
[Issue] https://github.com/Samsung/TAU/issues/932
[Problem] Cannot input the value using the keypad
[Solution]
- Add input elements to enter the value by keypad
- Apply design style according to the guide
- Add new APIs to set the value of the input element and focus the input element.

Signed-off-by: Hunseop Jeong <hs85.jeong@samsung.com>

HunseopJeong committed:
TimePicker (mobile): Implement 4-15 TimePicker dialog(SIP)

[Issue] https://github.com/Samsung/TAU/issues/932
[Problem] Cannot input the value using the keypad
[Solution]
- Add input elements to enter the value by keypad
- Apply design style according to the guide
- Add new APIs to set the value of the input element and focus the input element.

Signed-off-by: Hunseop Jeong <hs85.jeong@samsung.com>

HunseopJeong commented:
![image](https://user-images.githubusercontent.com/29908741/79395673-208ad900-7fb5-11ea-8f1e-774d9cef8cac.png)

singa2000 commented on src/js/profile/mobile/widget/TimePicker.js:
```diff
@@ -261,8 +347,175 @@
+					if (!self._inputValueState) {
+						visibleValue = parseInt(diffValue(self._previousInputValue, value), 10);
+						self._setInputValue("hour", visibleValue, true);
+
```
If you input 14 hours by SIP in the 12h format,  do you show the toast error message??

HunseopJeong replied:
+418							// TODO: Show toast error message
I will add the toast popup when finishing the implementaion of the toast.

singa2000 commented on src/js/profile/mobile/widget/TimePicker.js:
```diff
@@ -146,20 +159,31 @@
+				spinInput.min = options.min;
+				spinInput.max = options.max;
+				spinInput.type = "number";
+				spinInput.step = "1";
```
What's mean the "step"?

HunseopJeong replied:
The step attribute specifies the legal number intervals for a number field.
Example: if step="3", legal numbers could be -3, 0, 3, 6, etc.

singa2000 commented on src/css/profile/mobile/common/timepicker.less:
```diff
@@ -47,5 +47,35 @@
+		justify-content: center;
+
+		.ui-time-picker-input {
+			width: 50 * @px_base;
```
I have a question.
The input  focused box size by SIP is defined in the guide??

HunseopJeong replied:
No. There is no guide for the background size.
I will check it whether the width and height can be removed.

HunseopJeong replied:
If I remove the width and height, 
![image](https://user-images.githubusercontent.com/29908741/79420238-7ed6ac80-7ff3-11ea-8289-eef822c5a8cb.png)
Background color fit with input element size.
Do I need to require the guide for the background size?

singa2000 replied:
Now it's OK. 
You don't need to change your implementation.
It's just my question.^^

In next time, let's ask it to UX team.

TomaszLukawskiSam commented on src/js/profile/mobile/widget/TimePicker.js:
```diff
@@ -146,20 +159,31 @@
+				spinInput.max = options.max;
+				spinInput.type = "number";
+				spinInput.step = "1";
+				spinInput.classList.add(TimePicker.classes.TIME_INPUT);
```
You can write just `classes.TIME_INPUT`

HunseopJeong replied:
I know it, but I used it intentionally. Because Spin classes are also used, I wanted to distinguish clearly. If it doesn't matter, I want to use it.

lslachciak reviewed (commented):
typo in commit message: 
elmeents  -> elements

TomaszLukawskiSam commented on src/css/profile/mobile/common/timepicker.less:
```diff
@@ -47,5 +47,35 @@
+			outline: unset;
+			outline-offset: unset;
+			text-shadow: 0 0 0 var(--primary-dark-color);
+			opacity: 0%;
```
As I remember the opacity in percents makes issue on Tizen emulator. I have to check this.

TomaszLukawskiSam replied:
Confirmed. The opacity has to be in range 0 to 1 without percent

HunseopJeong replied:
Do we have to change the opacity value to be in range 0 to 1?
Okay I will change it.

HunseopJeong commented:
> typo in commit message:
> elmeents -> elements

Done.

TomaszLukawskiSam reviewed (changes requested):
Widget works, but issue with `opacity` needs to be resolved.

HunseopJeong commented:
@singa2000, @DohyungLim, @TomaszLukawskiSam 
I updated code to remove the default padding of the popup content.
So I added the 'ui-popup-has-time-picker' class to unset the padding.
Could you review this PR again?
